index.html

```diff
@@ -0,0 +1,475 @@
+**Consumers** consume CSDL. This spec places requirements on csdl consumers.
+
+<a name=def-composer></a>
+**Composers** or **compilers** are producers which compose subgraph schemas into CSDL. This document places no particular requirements on the composition algorithm, except that it must produce valid csdl.
+
+<a namme=def-router></a>
+**Routers** are consumers which serve a composed schema as a GraphQL endpoint.
+  - Graph routers differ from standard GraphQL endpoints in that they are not expected to process data or communicate with (non-GraphQL) backend services on their own. Instead, graph routers receive GraphQL requests and service them by performing additional GraphQL requests. This spec provides guidance for implementing routers, but does not require particular implementations of query separation or dispatch.
+  - Routers **may**—and typically will—omit CSDL directives and types from the schema they present to clients via introspection
+
+<a name=def-endpoint></a>
+**Endpoints** are running servers which can resolve GraphQL queries against a schema. In this version of the spec, endpoints must be URLs, typically http/https URLs.
+
+<a name=def-subgraph></a>
+**Subgraphs** are the GraphQL schemas which were composed to form the CSDL. A subgraph has:
+  - a name, which must be unique within the CSDL
+  - an [endpoint](#endpoint)
+
+This spec does not place any requirements on subgraph schemas. Generally, they may be of any shape. In particular, subgraph schemas do not need to provide a CSL; neither is it an error for them to do so. Composers **may** place additional requirements on subgraph schemas to aid in composition; composers **should** document any such requirements.
+
+## Structure
+
+A CSDL document must be a valid [GraphQL schema definition language](https://spec.graphql.org/draft/#sec-Type-System) document.
+
+CSDL removes certain SDL features, as described in the [divergence](#divergence-from-graphql-sdl) section. Nevertheless, CSDL consumers **should** be prepared to parse any valid GraphQL SDL. A CSDL document which contains disallowed SDL constructs **should** trigger validation errors after parsing.
+
+CSDL introduces a suite of types and directives. All CSDL-specific types and directives are prefixed with `csdl_` as recommended in [the GraphQL specification](https://spec.graphql.org/draft/#note-ca863).
+
+### Using this spec
+
+CSDL documents **must** contain a `schema` declaration with a [`@using` directive](/using/draft/pre-0).
+
+```graphql
+#::[`@using` directive declaring that the document conforms to csdl v1]
+schema @using(spec: "https://specs.apollo.dev/csdl/v1")
+```
+
+### Forbidden GraphQL IDL features
+
+CSDL documents cannot contain certain [GraphQL type system](https://spec.graphql.org/draft/#sec-Type-System) constructs. Specifically, **extensions are forbidden.** You cannot `extend type` or `extend interface` within a CSDL document. Producers **must** produce documents without extensions.
```

Comment:
While it's a good idea to reiterate this explicitly, note that my suggestion above also implies this by construction rather than by inclusion.

Reply:
Agree here that we should just say that CSDL documents can only consist of `TypeSystemDefinition`s by construction.

Reply:
If we decide to go with keys as fragments, that would break this assumption. Including both `TypeSystemDefinition`s and `FragmentDefinition`s is legal according to the grammar (there is only one type of GraphQL document and it can mix type system and executable definitions), but depending on where we expect to use CSDL we may want to check whether all implementations can handle this.